workitem_blackbox_test.go

```diff
@@ -89,6 +91,39 @@ func TestGetWorkItemWithLegacyDescription(t *testing.T) {
 	test.DeleteWorkitemOK(t, nil, nil, controller, *result.Data.ID)
 }
 
+func TestReorderWorkItem(t *testing.T) {
+	resource.Require(t, resource.Database)
+	pub, _ := almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	priv, _ := almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	svc := testsupport.ServiceAsUser("TestGetWorkItem-Service", almtoken.NewManager(pub, priv), testsupport.TestIdentity)
+	assert.NotNil(t, svc)
+	controller := NewWorkitemController(svc, gormapplication.NewGormDB(DB))
+	assert.NotNil(t, controller)
+	payload1 := minimumRequiredCreateWithType(workitem.SystemBug)
+	payload1.Data.Attributes[workitem.SystemTitle] = "Reorder Test WI"
+	payload1.Data.Attributes[workitem.SystemState] = workitem.SystemStateClosed
+
+	_, result1 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	_, result2 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	_, result3 := test.CreateWorkitemCreated(t, svc.Context, svc, controller, &payload1)
+	payload2 := minimumRequiredReorderPayload()
+	var dataArray []*app.WorkItem2
+	dataArray = append(dataArray, result2.Data, result3.Data)
+	payload2.Data = dataArray
+	payload2.Position.Above = *result1.Data.ID
+
+	_, reordered1 := test.ReorderWorkitemOK(t, nil, nil, controller, &payload2)
+	assert.NotNil(t, reordered1.Data)
```

Comment:
This must be `require.NotNil` or else we might run into a runtime panic due to null pointer dereference below. Let's try to avoid those panics in tests.

You might also want to add a required length check of the `Data` array because you're "blindly" accessing the first and second element below. `require.Len(t, 2, reordered1.Data)` should do the trick.